### PR TITLE
Packet storage for relay#15#16

### LINKED ISSUE #15: Store unrelayed AX.25 packets locally
Time estimate: 12h

### LINKED ISSUE #16: Relay unrelayed packets when connection is restored
Time estimate: 8h

### PR DESCRIPTION
**Pull request by teet71**
closes #15  closes #16

### CODE REVIEW

**Review by Risto-R — APPROVED** [review]

File: client/configuration.ini.sample [hunk 1]
```diff
@@ -6,6 +6,8 @@ mcs-relay-url = http://staging.estcube.eu:8029/ax.25
 # Type of the SIDS relay request. Possible values are GET and POST.
 # (Currently only POST is supported by ESTCube servers).
 relay-request-type = POST
+#
+save-unrelayed-packets = True
```
Comment (nowittyusername) [inline client/configuration.ini.sample]: Why is this an option in the .ini file? It should do it by default anyways.

Comment (nowittyusername) [reply client/configuration.ini.sample]: Secondly, there is no comments about what this enables or disables.

File: client/src/api.py [hunk 2]
```diff
@@ -57,6 +57,15 @@ def create_app(config: Configuration, tnc_pool: TNCPool, sids_relay: SIDSRelay)
     def get_sids_status():
         return jsonify(sids_relay.get_status()), 200
 
+    @app.route("/api/sids/toggle", methods=["POST"])
+    def toggle_relay():
+        some_json = request.get_json()
```
Comment (nowittyusername) [inline client/src/api.py]: Why is this called "some_json"? This isn't really descriptive of what this json really holds within. I would strongly suggest thinking of a better name.

File: client/src/api.py [hunk 3]
```diff
@@ -57,6 +57,15 @@ def create_app(config: Configuration, tnc_pool: TNCPool, sids_relay: SIDSRelay)
...
+        value = some_json['Mission Control']['relay-enabled']
```
Comment (nowittyusername) [inline client/src/api.py]: What "value"? In my opinion it should be like "current_relay_status" or something like that... Change!

File: client/src/conf.py [hunk 4]
```diff
@@ -8,6 +8,11 @@
 
 CONSTRAINTS = {
     "Mission Control": {
+        "save-unrelayed-packets": {
+            "type": "bool",
+            "label": "Save unrelayed packets for relay when connection is restored",
```
Comment (nowittyusername) [inline client/src/conf.py]:
```suggestion
            "label": "Save unrelayed packets for eventual forwarding to mission control when connection is restored",
```

File: client/src/db_interface.py [hunk 5]
```diff
@@ -37,7 +37,7 @@ def init_db(self):
         self._logger.info("Initializing database at %s", self.conn_str)
         conn = apsw.Connection(self.conn_str)
         cur = conn.cursor()
-        cur.execute("create table if not exists ax_frame (time text, data blob);")
+        cur.execute("create table if not exists ax_frame (time text, data blob, needsRelay bit);")
```
Comment (nowittyusername) [inline client/src/db_interface.py]: I feel like this "create table if not exists ax_frame" text could be changed into something more readable.
Secondly, we use snake_case in Python codem "needsRelay" isn't snake_case!

File: client/src/sids_relay.py [hunk 6]
```diff
@@ -44,7 +54,7 @@ def relay(self, frame: AXFrame):
         params = {
             "noradID": self.config.get_conf("Mission Control", "norad-id"),
             "source": self.config.get_conf("Mission Control", "receiver-callsign"),
-            "timestamp": frame.recv_time.isoformat(),
+            "recvTime": frame.recv_time.isoformat(),
```
Comment (nowittyusername) [inline client/src/sids_relay.py]: What's the reasoning here for changing this variable? The actual spec sheet calls it "timestamp" too. https://picsat.obspm.fr/sids/doc?locale=en

Secondly, the variable name isn't in snake_case!

**Review by nowittyusername — CHANGES_REQUESTED** [review]
This needs further work. Please adress my questions

File: client/src/sids_relay.py (hunk 6 above)
Comment (teet71) [reply client/src/sids_relay.py]: The endpoint excepted a value called recvTime so I changed it. I gather that we should change the endpoint instead.

Comment (nowittyusername) [reply client/src/sids_relay.py]: Yes.

File: client/configuration.ini.sample (hunk 1 above)
Comment (teet71) [reply client/configuration.ini.sample]: Removed the option from the configuration - it is now always done by default.

File: client/src/db_interface.py (hunk 5 above)
Comment (nowittyusername) [reply client/src/db_interface.py]: Disregard the first section of this comment.

**Review by nowittyusername — APPROVED** [review]

**Pull request merged**